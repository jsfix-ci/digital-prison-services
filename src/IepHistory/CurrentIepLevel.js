import React from 'react'
import PropTypes from 'prop-types'
import { H3 } from '@govuk-react/heading'
import GridRow from '@govuk-react/grid-row'
import GridCol from '@govuk-react/grid-col'
import DateFormatter from '../DateFormatter'

const CurrentIepLevel = ({ level, days }) => (
  <React.Fragment>
    <GridRow>
      <GridCol setWidth="one-quarter">
        <H3>Current IEP level</H3>
      </GridCol>
      <GridCol setWidth="one-quarter">
        <H3>Days since review</H3>
      </GridCol>
      <GridCol setWidth="one-quarter">
        <H3>Date of next review</H3>
      </GridCol>
    </GridRow>
    <GridRow>
      <GridCol setWidth="one-quarter">{level}</GridCol>
      <GridCol setWidth="one-quarter">{days}</GridCol>
      <GridCol setWidth="one-quarter" />
    </GridRow>
  </React.Fragment>
)

CurrentIepLevel.propTypes = {
  level: PropTypes.string.isRequired,
  days: PropTypes.number.isRequired,
}

export default CurrentIepLevel
