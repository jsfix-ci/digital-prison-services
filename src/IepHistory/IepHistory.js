import React from 'react'
import PropTypes from 'prop-types'
import Table from '@govuk-react/table'

const IepHistory = ({ iepHistory }) => (
  <Table caption="IEP history">
    <Table.Row>
      <Table.CellHeader>Date and Time</Table.CellHeader>
      <Table.CellHeader>IEP Description</Table.CellHeader>
      <Table.CellHeader>Comments</Table.CellHeader>
      <Table.CellHeader>Establishment</Table.CellHeader>
      <Table.CellHeader>Staff member</Table.CellHeader>
    </Table.Row>
    {iepHistory.map(row => (
      <Table.Row key={row.iepTime}>
        <Table.Cell>{row.formattedTime}</Table.Cell>
        <Table.Cell>{row.iepLevel}</Table.Cell>
        <Table.Cell>{row.comments || '\u2014'}</Table.Cell>
        <Table.Cell>{row.iepEstablishment}</Table.Cell>
        <Table.Cell>{row.iepStaffMember || '\u2014'}</Table.Cell>
      </Table.Row>
    ))}
  </Table>
)

IepHistory.propTypes = {
  iepHistory: PropTypes.arrayOf(
    PropTypes.shape({
      iepDescription: PropTypes.string.isRequired,
      comments: PropTypes.string,
      iepEstablishment: PropTypes.string.isRequired,
      iepStaffMember: PropTypes.string,
      iepTime: PropTypes.string.isRequired,
      formattedTime: PropTypes.string.isRequired,
    })
  ),
}
IepHistory.defaultProps = {
  iepHistory: [],
}

export default IepHistory
