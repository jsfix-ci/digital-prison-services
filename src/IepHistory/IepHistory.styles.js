import styled from 'styled-components'
import { MEDIA_QUERIES, SPACING } from '@govuk-react/constants'
import Select from '@govuk-react/select'
import { GREY_3, WHITE } from 'govuk-colours'
import GridCol from '@govuk-react/grid-col'
import { H3 } from '@govuk-react/heading'
import { typography } from '@govuk-react/lib'

export const CurrentIepLevelArea = styled.div`
  margin-bottom: ${SPACING.SCALE_4};
`
export const FullWidthSelect = styled(Select)`
  select {
    width: 100% !important;
    background-color: ${WHITE};
  }
`

export const SearchArea = styled.div`
  background: ${GREY_3};
  padding: ${SPACING.SCALE_5};
  margin-bottom: ${SPACING.SCALE_3};
`

export const LargeScreenOnlyGridCol = styled(GridCol)`
  display: none;
  ${MEDIA_QUERIES.TABLET} {
    display: block;
    height: 100%;
  }
`
export const FiltersLabel = styled(H3)`
  padding-left: ${SPACING.SCALE_2};
  margin-bottom: ${SPACING.SCALE_2};
  margin-top: -${SPACING.SCALE_2};
`
export const DateRangeLabel = styled.span`
  padding-left: ${SPACING.SCALE_2};
  ${typography.font({ size: 19 })};
  font-weight: bold;
  margin-top: ${SPACING.SCALE_1};
`
