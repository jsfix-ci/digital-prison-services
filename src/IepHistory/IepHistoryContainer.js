import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import moment from 'moment'

import Page from '../Components/Page'
import IepHistory from './IepHistory'
import CurrentIepLevel from './CurrentIepLevel'
import IepHistoryForm from './IepHistoryForm'
import { setIepHistoryResults, setIepHistoryFilter} from '../redux/actions'

const axios = require('axios')

class IepHistoryContainer extends Component {
  componentDidMount() {
    this.getIepHistory({})
  }

  componentDidUpdate = prevProps => {
    const { offenderNo } = this.props

    if (offenderNo !== prevProps.offenderNo) {
      this.getIepHistory({})
    }
  }

  getIepHistory = async ({ establishment, level, fromDate, toDate }) => {
    const {
      offenderNo,
      handleError,
      resetErrorDispatch,
      setLoadedDispatch,
      setIepHistoryResults: setResults,
    } = this.props

    resetErrorDispatch()
    setLoadedDispatch(false)

    try {
      const { data: results, headers } = await axios.get(`/api/bookings/${offenderNo}/iepSummary`)

      setResults({ ...results })
    } catch (error) {
      handleError(error)
    }

    setLoadedDispatch(true)
  }

  applyFilter = async formValues => {
    this.updateResults({ ...formValues })
  }

  updateResults = fields => {
    const { setIepHistoryFilter: setFilter } = this.props
    setFilter(fields)
    return this.getIepHistory(fields)
  }

  reset = () => this.updateResults({ establishment: null, level: null, fromDate: null, toDate: null })

  render() {
    return (
      <Page title="">
        <CurrentIepLevel />
        <IepHistoryForm search={this.applyFilter} reset={this.reset} />
        <IepHistory />
      </Page>
    )
  }
}

IepHistoryContainer.propTypes = {
  offenderNo: PropTypes.string.isRequired,
  handleError: PropTypes.func.isRequired,
  setIepHistoryResults: PropTypes.func.isRequired,
  setIepHistoryFilter: PropTypes.func.isRequired,
  // history from Redux Router Route
  history: PropTypes.shape({
    replace: PropTypes.func.isRequired,
  }).isRequired,
  fieldValues: PropTypes.shape({
    establishment: PropTypes.string,
    level: PropTypes.string,
    fromDate: PropTypes.instanceOf(moment),
    toDate: PropTypes.instanceOf(moment),
  }).isRequired,

  // redux dispatch
  resetErrorDispatch: PropTypes.func.isRequired,
  setLoadedDispatch: PropTypes.func.isRequired,
}

const mapStateToProps = state => ({
  error: state.app.error,
  fieldValues: {
    establishment: state.iepHistory.establishment,
    level: state.iepHistory.level,
    fromDate: state.iepHistory.fromDate,
    toDate: state.iepHistory.toDate,
  },
})

const mapDispatchToProps = {
  setIepHistoryResults,
  setIepHistoryFilter,
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(IepHistoryContainer)
