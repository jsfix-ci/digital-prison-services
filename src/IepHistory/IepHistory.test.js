import React from 'react'
import { shallow } from 'enzyme'
import IepHistory from './IepHistory'

describe('IEP History', () => {
  it('should render the page correctly', () => {
    const iepHistory = [
      {
        bookingId: -1,
        iepDate: '2017-08-15',
        iepTime: '2017-08-15T16:04:35',
        formattedTime: '15/08/2017 - 16:04',
        agencyId: 'LEI',
        iepLevel: 'Standard',
        comments: 'Test comment.',
      },
    ]
    const wrapper = shallow(<IepHistory iepHistory={iepHistory} />)

    expect(wrapper).toMatchSnapshot()
  })
})
