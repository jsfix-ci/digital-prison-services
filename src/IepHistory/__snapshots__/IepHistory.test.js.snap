// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IEP History should render the current IEP level correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(CurrentIepLevel)
    store={
      Object {
        "dispatch": [MockFunction],
        "getState": [MockFunction] {
          "calls": Array [
            Array [],
            Array [],
          ],
        },
        "subscribe": [MockFunction] {
          "calls": Array [
            Array [
              [Function],
            ],
          ],
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "days": 625,
      "dispatch": [MockFunction],
      "level": "Standard",
      "nextReviewDate": "15/08/2018",
      "store": Object {
        "dispatch": [MockFunction],
        "getState": [MockFunction] {
          "calls": Array [
            Array [],
            Array [],
          ],
        },
        "subscribe": [MockFunction] {
          "calls": Array [
            Array [
              [Function],
            ],
          ],
        },
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [MockFunction],
          "getState": [MockFunction] {
            "calls": Array [
              Array [],
              Array [],
            ],
          },
          "subscribe": [MockFunction] {
            "calls": Array [
              Array [
                [Function],
              ],
            ],
          },
        },
        "unsubscribe": undefined,
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "days": 625,
        "dispatch": [MockFunction],
        "level": "Standard",
        "nextReviewDate": "15/08/2018",
        "store": Object {
          "dispatch": [MockFunction],
          "getState": [MockFunction] {
            "calls": Array [
              Array [],
              Array [],
            ],
          },
          "subscribe": [MockFunction] {
            "calls": Array [
              Array [
                [Function],
              ],
            ],
          },
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "dispatch": [MockFunction],
            "getState": [MockFunction] {
              "calls": Array [
                Array [],
                Array [],
              ],
            },
            "subscribe": [MockFunction] {
              "calls": Array [
                Array [
                  [Function],
                ],
              ],
            },
          },
          "unsubscribe": undefined,
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": true,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): Object {
    "storeSubscription": undefined,
  },
}
`;

exports[`IEP History should render the iep history table correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(IepHistory)
    store={
      Object {
        "dispatch": [MockFunction],
        "getState": [MockFunction] {
          "calls": Array [
            Array [],
            Array [],
          ],
        },
        "subscribe": [MockFunction] {
          "calls": Array [
            Array [
              [Function],
            ],
          ],
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "dispatch": [MockFunction],
      "results": Array [
        Object {
          "agencyId": "LEI",
          "bookingId": -1,
          "formattedTime": "13/08/2017 - 16:04",
          "iepDate": "2017-08-13",
          "iepEstablishment": "Leeds",
          "iepLevel": "Standard",
          "iepStaffMember": "Staff Member",
          "iepTime": "2017-08-13T16:04:35",
          "userId": "ITAG_USER",
        },
      ],
      "store": Object {
        "dispatch": [MockFunction],
        "getState": [MockFunction] {
          "calls": Array [
            Array [],
            Array [],
          ],
        },
        "subscribe": [MockFunction] {
          "calls": Array [
            Array [
              [Function],
            ],
          ],
        },
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [MockFunction],
          "getState": [MockFunction] {
            "calls": Array [
              Array [],
              Array [],
            ],
          },
          "subscribe": [MockFunction] {
            "calls": Array [
              Array [
                [Function],
              ],
            ],
          },
        },
        "unsubscribe": undefined,
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "dispatch": [MockFunction],
        "results": Array [
          Object {
            "agencyId": "LEI",
            "bookingId": -1,
            "formattedTime": "13/08/2017 - 16:04",
            "iepDate": "2017-08-13",
            "iepEstablishment": "Leeds",
            "iepLevel": "Standard",
            "iepStaffMember": "Staff Member",
            "iepTime": "2017-08-13T16:04:35",
            "userId": "ITAG_USER",
          },
        ],
        "store": Object {
          "dispatch": [MockFunction],
          "getState": [MockFunction] {
            "calls": Array [
              Array [],
              Array [],
            ],
          },
          "subscribe": [MockFunction] {
            "calls": Array [
              Array [
                [Function],
              ],
            ],
          },
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "dispatch": [MockFunction],
            "getState": [MockFunction] {
              "calls": Array [
                Array [],
                Array [],
              ],
            },
            "subscribe": [MockFunction] {
              "calls": Array [
                Array [
                  [Function],
                ],
              ],
            },
          },
          "unsubscribe": undefined,
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": true,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): Object {
    "storeSubscription": undefined,
  },
}
`;
